Public/Sdk/Public/Managed/Testing/XUnit/xunit.dsc

```diff
@@ -65,6 +65,14 @@ export function runConsoleTest(args: TestRunArguments): Result {
 
     const result = Transformer.execute(execArguments);
 
+    if (Environment.hasVariable("BUILDXL_IS_IN_CLOUDBUILD")) {

```

Comment:
The `"BUILDXL_IS_IN_CLOUDBUILD"` string is also used in `qtestFramework.dsc` and `Tool.QTestRunner.dsc`, so maybe it would be worth moving it to `BuildXLSdk.dsc` and adding something like
```typescript
@@public export const isInCloudBuild = Environment.hasVariable("BUILDXL_IS_IN_CLOUDBUILD");
```

Reply:
I know it's convenient but I'm not a big fan of our "is in cloudbuild" type of flags. I'd rather us tie this sort of thing to the capabilities needed, not the system and our current understanding of its capabilities are. Maybe in 10 years we'll be be more advanced than the cloud and be running in spacebuild.

At the very least please comment why the test needs to behave differently and what assumptions it's making about the environment here

Reply:
I was thinking of the same thing. However, we cannot pull BuildXLSdk.dsc here because it means that the SDK that we publish to customers will depend on BuildXLSdk.dsc.

Either
- have a baked Environment.isInCloudBuild(), or
- add in process pip "logOutput: " ... similar to "output", but it goes to log folder.
For sure, both should not be in this PR.

---
In reply to: [293865962](https://github.com/microsoft/BuildXL/pull/457#discussion_r293865962) [](ancestors = 293865962)

Reply:
I think it would be a nice behavior to copy the xunit logs to the build session's log directory regardless of whether it's running in CB or not. Do you see problems with that?

Then there's how to implement it. Since we now have a log mount it could be exposed that way and have build logic authors do it in DScript. Or we could have an annotation on the pips that some files are outputs and thus should be handled special. What do you think?

Reply:
I think it's aligned to what I replied to Aleks, and what I mentioned in the user story. I want to have `Context.getNewOutputLogDirectory(...)`, which is similar to `Context.getNewOutputDirectory(...)`, but rooted at the log directory, instead of at the object directory.

---
In reply to: [294013891](https://github.com/microsoft/BuildXL/pull/457#discussion_r294013891) [](ancestors = 294013891)